### PR TITLE
fix: setter mutation issue

### LINKED ISSUE #25364: [Bug]: Calling the setDisabled method of the component resets the values already assigned by the JS script.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I used the setter method for the component, I found that the values assigned by the previous variable were reset.

### Steps To Reproduce

1.Drag an input onto the canvas.
2.Create JS scripts.
3.JS scripts Code.
export default {
	myVar1: [],
	myVar2: {},
	myVar:'',
	myFun1 () {
		this.myVar = 'Hello'
		Input1.setDisabled(false)
		console.log(this.myVar) 
	},
}
4.The runtime observed the value of the variable myVar and found that myVar was reset.

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Appsmith v1.9.27-SNAPSHOT

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

When we added a mutation task to the microtask queue, earlier we didn't update the evalProps as soon as the change is triggered, due to this we lost the update. 

Now, we update the evalProps as soon as the setter or getter for JSObject variable is triggered making sure the latest value is always present. 

#### PR fixes following issue(s)
Fixes #25364

#### Type of change

- Bug fix (non-breaking change which fixes an issue)

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/workers/Evaluation/evalTreeWithChanges.ts [hunk 1]
```diff
@@ -65,7 +65,10 @@ export function evalTreeWithChanges(
       evalProps: dataTreeEvaluator.evalProps,
     });
 
-    evalMetaUpdates = [...evalMetaUpdates, ...updateResponse.evalMetaUpdates];
+    /** Make sure evalMetaUpdates is sanitized to prevent postMessage failure */
+    evalMetaUpdates = JSON.parse(
+      JSON.stringify([...evalMetaUpdates, ...updateResponse.evalMetaUpdates]),
+    );
```
Comment (arunvjn) [inline app/client/src/workers/Evaluation/evalTreeWithChanges.ts:71]: Why do we have to stringify and parse here?

Comment (Rishabh-Rathod) [reply app/client/src/workers/Evaluation/evalTreeWithChanges.ts:71]: `evalMetaUpdates` stores evaluated values that could possibly fail the structuredClone while sending postMessage. 

We were doing this earlier too, it just got missed while implementing the widget property setter feature.

**Review by arunvjn — APPROVED** [review]

**Pull request merged**